include/picongpu/fields/laserProfiles/PulseFromSpectrum.hpp

```diff
@@ -0,0 +1,335 @@
+                    static constexpr float_X FOCUS_POS = float_X(Params::FOCUS_POS_SI / UNIT_LENGTH); // unit: meter
+                    static constexpr float_X INIT_TIME = float_X(
+                        (Params::PULSE_INIT * Params::PULSE_DURATION_SI)
+                        / UNIT_TIME); // unit: seconds (full inizialisation duration)
+
+                    /* Rayleigh-Length in y-direction
+                     */
+                    static constexpr float_X Y_R = float_X(PI * W0 * W0 / WAVE_LENGTH); // unit: meter
+                };
+            } // namespace pulseFromSpectrum
+
+            namespace acc
+            {
+                template<typename T_Unitless>
+                struct PulseFromSpectrum
+                    : public T_Unitless
+                    , public acc::BaseFunctor<T_Unitless::initPlaneY>
+                {
+                    using Unitless = T_Unitless;
+                    using BaseFunctor = acc::BaseFunctor<T_Unitless::initPlaneY>;
+
+                    float3_X m_elong;
+                    float_X m_phase;
+                    float_X m_currentStep;
+                    typename FieldE::DataBoxType m_dataBoxE;
+                    DataSpace<simDim> m_offsetToTotalDomain;
+                    DataSpace<simDim> m_superCellToLocalOriginCellOffset;
+
+                    /** waist of the laser beam
+                     * @param y_to_focus y-coordinate relative to focus-position
+                     */
+                    HDINLINE float_X waist(float_X y_to_focus)
+                    {
+                        return Unitless::W0 * math::sqrt(1.0 + math::pow(y_to_focus / Unitless::Y_R, 2.0));
+                    }
+
+                    /** inverse radius of the beam-curvature
+                     * @param y_to_focus y-coordinate relative to focus-position
+                     */
+                    HDINLINE float_X R_inv(float_X y_to_focus)
+                    {
+                        float_X y_R = Unitless::Y_R; // needed since pow() requires lvalue
+                        return y_to_focus / (math::pow(y_to_focus, 2.0) + math::pow(y_R, 2.0));
+                    }
+
+                    /** gauss spectrum
+                     * v0 is the central frequency
+                     * PULSE_DURATION is sigma of std. gauss for intensity (E^2) in time domain
+                     * @param v frequency
+                     */
+                    HDINLINE float_X gauss_spectrum(float_X v)
+                    {
+                        // Norm is choosen so that the maximum after fourier transformation is unity.
+                        float_X norm = math::sqrt(float_X(PI)) * Unitless::PULSE_DURATION;
+                        return norm
+                            * math::exp(
+                                   -1.0
+                                   * math::pow(
+                                       2._X * Unitless::PULSE_DURATION * float_X(PI) * (v - Unitless::v0),
```

Comment:
Why is this mixed-precision? Perhaps `-math::exp(...)` ?

Reply:
Refactored